src/modules/hero/hero.component.ts

```diff
@@ -15,5 +16,43 @@ export class StacheHeroComponent {
     return this._backgroundImageUrl;
   }
 
+  @Input()
+  set overlayOpacity(opacityPercentage: string) {
+    let validOpacityInput = opacityPercentage.replace(/[^\d.-]/g, '');
+    let validOpacityAsString = this.parseOpacityValueToDecimal(validOpacityInput);
+    this._overlayOpacity = validOpacityAsString;
+  }
+
+  get overlayOpacity(): string {
+    return this._overlayOpacity;
+  }
+
   private _backgroundImageUrl = 'none';
+  private _overlayOpacity = '0.6';
+
+  private parseOpacityValueToDecimal(opacityValue: string): string {
+    let percentageCandiate = parseInt(opacityValue, 10);
+    if (opacityValue.includes('.')) {
+      percentageCandiate = Math.round(parseFloat(opacityValue) * 100) / 100;
+    }
+    let validDecimal = this.validateOpacityDecimal(percentageCandiate);
+    return String(validDecimal);
+  }
+
+  private validateOpacityDecimal(value: number): number {
```

Comment:
I was initially confused by the naming of this method and had to read through it to figure out what it was doing. It's really just parsing a number into an [interval](https://en.wikipedia.org/wiki/Interval_(mathematics)). What about...

```
// Converts a number to a value between 0 and 1.
private parseInterval(value: number): number {
  let interval = parseFloat(value);

  if (isNaN(interval)) {
    return 0.4;
  }

  if (interval > 100) {
    return 1;
  }

  if (interval < 0) {
    return 0;
  }

  if (interval % 1 > 0) {
    return interval;
  }

  return interval / 100;
}
```